mate3/field_values.py

```diff
@@ -0,0 +1,177 @@
+
+    @property
+    def scale_factor(self) -> int:
+        if self._last_read is None:
+            raise RuntimeError("Can't access 'scale_factor' before the field value has been read at least once.")
+        return self._scale_factor
+
+    @property
+    def raw_value(self) -> Any:
+        if self._last_read is None:
+            raise RuntimeError("Can't access 'raw_value' before the field value has been read at least once.")
+        return self._raw_value
+
+    @property
+    def _should_be_scaled(self):
+        return isinstance(self.field, IntegerField) and self.field.scale_factor is not None
+
+    @property
+    def value(self) -> Any:
+        if self._last_read is None:
+            raise RuntimeError("Can't access 'value' before the field value has been read at least once.")
+        if self.field.mode not in (Mode.R, Mode.RW):
+            raise RuntimeError("Can't read from this field!")
+        if not self._implemented:
+            return None
+        if not self._should_be_scaled:
+            return self._raw_value
+        # OK, should be scaled, so let's scale it:
+        value = self._raw_value * 10 ** self._scale_factor
+        # Round it to what it should be after scaling:
+        return round(value, -self._scale_factor if self._scale_factor < 0 else 0)
+
+    @value.setter
+    def value(self, value):
+        # TODO: ensure the type of value is correct
+        if self.field.mode not in (Mode.W, Mode.RW):
+            raise RuntimeError("Can't write to this field!")
+        if self._last_read is None:
+            raise RuntimeError("You should read a field at least once before writing")
+        if not self._implemented:
+            raise RuntimeError("This field is marked as not implemented, so you shouldn't write to it!")
+        if self._should_be_scaled:
+            # Time limit on scale_factor being applicable?
+            if (datetime.now() - self._last_read) > self._scale_factor_cache_time:
+                raise ValueError(
+                    (
+                        f"You need to read this value within {self._scale_factor_cache_time} of writing to 'ensure'",
+                        " the scale factor is up-to-date before you write.",
+                    )
+                )
+            # Scale it:
+            value = value / 10 ** self._scale_factor
+            # Round it to what it should be after scaling:
+            # TODO: raise error if too many digits specified
+            self._value_to_write = int(round(value, 0))
+        else:
+            self._value_to_write = value
+        self._dirty = True
+
+    def _update_on_read(self, value, implemented, read_time, scale_factor_read=None):
+        """
+        Assumption is that if there's a scale factor, it's read at the same time as value, so they should be in sync.
+        Not really a major with Outback, as the scale factors seem to be constant anyway ... but no harm in being safe.
+        """

```

Comment:
Minor comment – Can this docstring start by saying what the method does?

Reply:
Done!